### DISCUSSION TITLE
Available contexts by default in workflow

### DISCUSSION THREAD

**Discussion by rhysd** [community/community#35565] [Actions] [answered]
https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability

This document describes which contexts are available for specific workflow keys. How about other keys which are not described here? I guess it is 'no context allowed' or 'all contexts allowed', but I could not understand which is correct from the document.

**Comment by antonblr** [comment, marked as answer]
Hi @rhysd - This is a comprehensive list, that is, if some workflow keys are missing, you can assume that contexts are not available for them.

**Reply by rhysd** [reply]
Thanks! It was made clear by your answer.

**Comment by ChristopherHX** [comment]
FYI The effective context availability of action.yml is described here https://github.com/actions/runner/blob/main/src/Runner.Worker/action_yaml.json. GitHub doesn't provide any document for action context availability in the docs.

Workflows are parsed by the actions service, therefore the workflow level https://github.com/actions/runner/blob/main/src/Sdk/DTPipelines/workflow-v1.0.json is barely maintained and doesn't cover the current workflow schema. The mentioned docs doesn't 100% reflect the current availability, they approuved a PR of invalid docs updates, e.g. the container property doesn't allow env.
The vars context will be introduced soon, the actions/runner was already updated to prepare rollout.

The default is 'no context allowed', which means every expression other than escaping expression syntax like `${{ '${{Test}}' }}` / `${{ 'somerandomstring' }}` is rejected. This isn't based on any docs, only trial and error.
Context Availability is inherited by downstream keys and values.

**Reply by rhysd** [reply]
Thanks! I didn't think about checking the runner implementation directly. The config file looks very useful to understand the behavior.
